File: SortWare/Interfaces/FileConversion.vb
Public Class FileConversion

    Private conversionDirs As List(Of SortDirectory)
    Private finishedDir As SortDirectory
    Public Sub New(ByVal convDirs As List(Of SortDirectory), ByVal finDir As SortDirectory)
        ' This call is required by the designer.
        InitializeComponent()

        conversionDirs = convDirs
        finishedDir = finDir
        refreshDirs()
    End Sub

    Private Sub refreshDirs()
        ConversionFolders.Items.Clear()

        For Each e In conversionDirs
            Dim lvi As New ListViewItem({e.getName, e.fullName, e.getScriptPath})
            ConversionFolders.Items.Add(lvi)
        Next
    End Sub

    Private Sub refreshFiles()
        FilesToBeConverted.Items.Clear()

        If ConversionFolders.SelectedItems.Count = 1 Then
            Dim loc As String = ConversionFolders.SelectedItems.Item(0).SubItems(1).Text
            Dim fType As String = System.Text.RegularExpressions.Regex.Match(ConversionFolders.SelectedItems.Item(0).Text, "^\S{0,}").Value.ToLower
            For Each f In IO.Directory.GetFiles(loc)
                If fType.Contains(IO.Path.GetExtension(f).ToLower.Remove(0, 1)) Then
                    Dim newItem As ListViewItem = FilesToBeConverted.Items.Add(f)
                    Dim newName As String = IO.Path.GetFileNameWithoutExtension(f) & ".mp4"
                    If IO.File.Exists(finishedDir.fullName & "\" & newName) Then
                        newItem.BackColor = Color.DarkRed
                        newItem.ForeColor = Color.White
                    End If
                End If
            Next
        End If
    End Sub

    Private Sub ConversionFolders_SelectedIndexChanged(sender As Object, e As EventArgs) Handles ConversionFolders.SelectedIndexChanged
        refreshFiles()
    End Sub

    Private Sub ConvSelected_Click(sender As Object, e As EventArgs) Handles ConvSelected.Click
        For Each f As ListViewItem In FilesToBeConverted.SelectedItems()
            Dim oldName As String = f.Text
            Dim fname As String = IO.Path.GetFileNameWithoutExtension(f.Text) & ".mp4"
            Dim newName As String = finishedDir.fullName & "\" & fname
            Dim scriptLoc As String = ConversionFolders.SelectedItems.Item(0).SubItems(1).Text
            Dim scriptName As String = IO.Path.GetFileName(ConversionFolders.SelectedItems.Item(0).SubItems(2).Text)
            Dim proc As New Process
            Try
                Dim batDir As String = String.Format("D:\")
                proc = New Process
                proc.StartInfo.WorkingDirectory = IO.Path.GetDirectoryName(ConversionFolders.SelectedItems.Item(0).SubItems(1).Text)
                proc.StartInfo.FileName = scriptLoc
                proc.StartInfo.Arguments = """" & oldName & """" & " " & """" & newName & """"
                proc.StartInfo.CreateNoWindow = False
                Dim newProc As Process = Process.Start(scriptLoc & "\" & scriptName, proc.StartInfo.Arguments)
                newProc.WaitForExit()
            Catch ex As Exception

            End Try
        Next
        refreshFiles()
    End Sub

    Private Sub ClearDone_Click(sender As Object, e As EventArgs) Handles ClearDone.Click
        If ConversionFolders.SelectedItems.Count = 1 Then
            Dim loc As String = ConversionFolders.SelectedItems.Item(0).SubItems(1).Text
            Dim fType As String = System.Text.RegularExpressions.Regex.Match(ConversionFolders.SelectedItems.Item(0).Text, "^\S{0,}").Value.ToLower
            For Each f In IO.Directory.GetFiles(loc)
                If fType.Contains(IO.Path.GetExtension(f).ToLower.Remove(0, 1)) Then
                    Dim newItem As ListViewItem = FilesToBeConverted.Items.Add(f)
                    Dim newName As String = IO.Path.GetFileNameWithoutExtension(f) & ".mp4"
                    If IO.File.Exists(finishedDir.fullName & "\" & newName) Then
                        IO.File.Delete(f)
                    End If
                End If
            Next
        End If
        refreshFiles()
    End Sub
End Class

File: SortWare/Interfaces/FileConversion.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using System.Drawing;
using System.Linq;
using System.Text.RegularExpressions;
using System.Windows.Forms;

namespace SortWare
{

    public partial class FileConversion
    {

        private List<SortDirectory> conversionDirs;
        private SortDirectory finishedDir;
        public FileConversion(List<SortDirectory> convDirs, SortDirectory finDir)
        {
            // This call is required by the designer.
            InitializeComponent();

            conversionDirs = convDirs;
            finishedDir = finDir;
            refreshDirs();
        }

        private void refreshDirs()
        {
            ConversionFolders.Items.Clear();

            foreach (var e in conversionDirs)
            {
                var lvi = new ListViewItem(new[] { e.getName(), e.fullName(), e.getScriptPath(), e.getConvTitle() });
                ConversionFolders.Items.Add(lvi);
            }
        }

        private void refreshFiles()
        {
            FilesToBeConverted.Items.Clear();

            if (ConversionFolders.SelectedItems.Count == 1)
            {
                string loc = ConversionFolders.SelectedItems[0].SubItems[1].Text;
                string[] fType = Regex.Matches(Regex.Match(ConversionFolders.SelectedItems[0].SubItems[3].Text, @"\S+").Value, "[^ -]+").OfType<Match>().Select((m) => m.Value.ToLower()).ToArray();
                foreach (var f in System.IO.Directory.GetFiles(loc))
                {
                    if (fType.Contains(System.IO.Path.GetExtension(f).ToLower().Remove(0, 1)))
                    {
                        var newItem = FilesToBeConverted.Items.Add(f);
                        newItem.SubItems.Add(Generics.GetFileSize(f));
                        string newName = System.IO.Path.GetFileNameWithoutExtension(f) + ".mp4";
                        if (System.IO.File.Exists(finishedDir.fullName() + @"\" + newName))
                        {
                            newItem.BackColor = Color.DarkRed;
                            newItem.ForeColor = Color.White;
                            newItem.SubItems.Add(Generics.GetFileSize(finishedDir.fullName() + @"\" + newName));
                        }
                    }
                }
            }
        }

        private void ConversionFolders_SelectedIndexChanged(object sender, EventArgs e)
        {
            refreshFiles();
        }

        private void ConvSelected_Click(object sender, EventArgs e)
        {
            foreach (ListViewItem f in FilesToBeConverted.SelectedItems)
            {
                string oldName = f.Text;
                string fname = System.IO.Path.GetFileNameWithoutExtension(f.Text) + ".mp4";
                string newName = finishedDir.fullName() + @"\" + fname;
                string scriptLoc = ConversionFolders.SelectedItems[0].SubItems[1].Text;
                string scriptName = System.IO.Path.GetFileName(ConversionFolders.SelectedItems[0].SubItems[2].Text);
                var proc = new Process();
                try
                {
                    string batDir = string.Format(@"D:\");
                    proc = new Process();
                    proc.StartInfo.WorkingDirectory = System.IO.Path.GetDirectoryName(ConversionFolders.SelectedItems[0].SubItems[1].Text);
                    proc.StartInfo.FileName = scriptLoc;
                    proc.StartInfo.Arguments = "\"" + oldName + "\"" + " " + "\"" + newName + "\"";
                    proc.StartInfo.CreateNoWindow = false;
                    var newProc = Process.Start(scriptLoc + @"\" + scriptName, proc.StartInfo.Arguments);
                    newProc.WaitForExit();
                }
                catch (Exception ex)
                {

                }
            }
            refreshFiles();
        }

        private void ClearDone_Click(object sender, EventArgs e)
        {
            if (ConversionFolders.SelectedItems.Count == 1)
            {
                string loc = ConversionFolders.SelectedItems[0].SubItems[1].Text;
                string[] fType = Regex.Matches(Regex.Match(ConversionFolders.SelectedItems[0].SubItems[3].Text, @"\S+").Value, "[^ -]+").OfType<Match>().Select((m) => m.Value.ToLower()).ToArray();
                foreach (var f in System.IO.Directory.GetFiles(loc))
                {
                    if (fType.Contains(System.IO.Path.GetExtension(f).ToLower().Remove(0, 1)))
                    {
                        var newItem = FilesToBeConverted.Items.Add(f);
                        string newName = System.IO.Path.GetFileNameWithoutExtension(f) + ".mp4";
                        if (System.IO.File.Exists(finishedDir.fullName() + @"\" + newName))
                        {
                            System.IO.File.Delete(f);
                        }
                    }
                }
            }
            refreshFiles();
        }
    }
}